shared/actions/config/index.js

```diff
@@ -67,7 +56,25 @@ export function getExtendedConfig (): AsyncAction {
   }
 }
 
-export function getCurrentStatus (): AsyncAction {
+let bootstrapSetup = false
+export function bootstrap (): AsyncAction {
+  return dispatch => {
+    if (!bootstrapSetup) {
+      bootstrapSetup = true
+      console.log('Registered bootstrap')
+      engine.listenOnConnect('bootstrap', () => {
+        console.log('Bootstrapping')
+        dispatch(bootstrap())
+      })
+    } else {
+      Promise.all([dispatch(getCurrentStatus()), dispatch(getExtendedStatus()), dispatch(getConfig())]).then(() => {
```

Comment:
i'm going to not do this now , per our conversation, as we'd be mixing which dispatch calls return a promise or not and this'll make the flow typing really wonky